pageserver/src/http/routes.rs

```diff
@@ -571,6 +571,47 @@ async fn tenant_status(request: Request<Body>) -> Result<Response<Body>, ApiErro
     )
 }
 
+async fn tenant_history_handler(request: Request<Body>) -> Result<Response<Body>, ApiError> {
+    let tenant_id: TenantId = parse_request_param(&request, "tenant_id")?;
+    check_permission(&request, Some(tenant_id))?;
+
+    let tenant = tokio::task::spawn_blocking(move || tenant_mgr::get_tenant(tenant_id, false))
+        .await
+        .map_err(|e: JoinError| ApiError::InternalServerError(e.into()))?
+        .map_err(ApiError::InternalServerError)?;
+
+    // this can be long operation, it currently is not backed by any request coalescing or similar
+    let inputs = tenant
+        .gather_pricing_inputs()
+        .await
+        .map_err(ApiError::InternalServerError)?;
+
+    let history_size = inputs.price().map_err(ApiError::InternalServerError)?;
+
+    /// Private response type with the additional "unstable" `inputs` field.
+    ///
+    /// The type is described with `id` and `history_size` in the openapi_spec file, but the
+    /// `inputs` is intentionally left out. The type resides in the pageserver not to expose
+    /// `ModelInputs`.
+    #[serde_with::serde_as]
+    #[derive(serde::Serialize)]
+    struct TenantHistorySize {
+        #[serde_as(as = "serde_with::DisplayFromStr")]
+        id: TenantId,
+        history_size: u64,
```

Comment:
What are the units of this size?

Reply:
I am pretty sure it's bytes given it's mix of WAL and logical size.

Reply:
After the `pricing_model` => `tenant_size` rename, this name changed to `size` I added a comment on it but now looking back, maybe you were thinking of encoding it to the name?

Reply:
> encoding it to the name

I think we've done it somewhere in the code already, but might be that the comment is also enough.